### PR TITLE
Update docs FAQ with component filtering

### PR DESCRIPTION
**Pull request by derrickmehaffy**
Signed-off-by: Derrick Mehaffy <derrickmehaffy@gmail.com>

### What does it do?

Updates the FAQ to clarify some information

### Why is it needed?

N/A

### Related issue(s)/PR(s)

N/A

### CODE REVIEW

File: docs/v3.x/getting-started/troubleshooting.md [hunk 1]
```diff
@@ -45,20 +45,20 @@ With the release of the Strapi beta version a fundamental change occurred in tha
...
 
 ### Relations aren't maintaining their sort order
 
 With the components there is a hidden field called `order` that allows entries to maintain their sort, however with relations there is no such field. If you consider the typical count of of component entries vs relational based entries (in retrospect they function in the backend the same) there is generally a much higher number of relations. If relations were to have an `order` field applied to them as well it could cause significant performance degradation when trying to update the order, and likewise in the case where a relation could be attached to multiple entries it would be quite difficult to maintain the order.
 
 For the time being there is no recommended way to handle this automatically and instead it may be required for you to create custom controllers to handle this within your own project.
 
-### Why is my app's database and uploads resetting on Heroku
+### Why is my app's database and uploads resetting on PaaS (Heroku, DigitalOcean Apps, Google App Engine, ect)
```
Comment (meganelacheny) [inline docs/v3.x/getting-started/troubleshooting.md:61]: We should avoid having parenthesis in a title, even more writing "etc".
I think you can stop after "Paas", but mention Heroku, DigitalOcean Apps and Google App Engine in the paragraph below.

File: docs/v3.x/getting-started/troubleshooting.md [hunk 2]
```diff
@@ -45,20 +45,20 @@ With the release of the Strapi beta version a fundamental change occurred in tha
...
 
 If you used `--quickstart` to create your Strapi project, by default this uses the SQLite database. Heroku's file system is [ephemeral](https://devcenter.heroku.com/articles/dynos#ephemeral-filesystem) meaning that each time a dyno (container) is reset all filesystem changes are lost. And since both SQLite and local uploads are stored on the filesystem, any changes made to these since the last dyno reset will be deleted. Typically dynos are reset at least once a day, and in most cases multiple times per day or when new code is pushed to Heroku.
```
Comment (meganelacheny) [inline docs/v3.x/getting-started/troubleshooting.md:63]: Is it normal that you wanted to mention DigitalOcean Apps and Google App Engine in the title, but here in the explanatory paragraph it's only about Heroku. I think it's confusing 🤔

File: docs/v3.x/getting-started/troubleshooting.md [hunk 3]
```diff
@@ -92,6 +92,10 @@ There is currently no ETA on this migration however.
 
 No, currently does not have the ability to allow for changing the default id name nor does it allow you to switch the data type (such as UUID on bookshelf and integer on mongoose), support for this is being looked at for Strapi v4.
 
+### Can you filter / deep filter on components, dynamic zones, and polymorphic relations
+
+Current it is not possible to filter on components, dynamic zones, or polymorphic relations. This is something we are looking into, if you want see more information please take a look at the following [GitHub issue](https://github.com/strapi/strapi/issues/5124).
```
Comment (meganelacheny) [inline docs/v3.x/getting-started/troubleshooting.md:97]:
```suggestion
Currently, it is not possible to filter on components, dynamic zones, or polymorphic relations. This is something we are looking into. If you want to see more information, please take a look at the following [GitHub issue](https://github.com/strapi/strapi/issues/5124).
```

**Review by meganelacheny — CHANGES_REQUESTED** [review]

File: docs/v3.x/getting-started/troubleshooting.md (hunk 1 above)
Comment (derrickmehaffy) [reply docs/v3.x/getting-started/troubleshooting.md:61]: Makes sense

**Review by meganelacheny — APPROVED** [review]

**Pull request merged**